rewrite-maven/src/main/java/org/openrewrite/maven/MavenVisitor.java

```diff
@@ -388,16 +392,21 @@ public boolean isDependencyLikeTag() {
     }
 
     public @Nullable Plugin findPlugin(Xml.Tag tag) {
-        List<Plugin> plugins = getResolutionResult().getPom().getPlugins();
-        if (plugins != null) {
-            for (Plugin resolvedPlugin : plugins) {
-                String reqGroup = resolvedPlugin.getGroupId();
-                String reqVersion = resolvedPlugin.getVersion();
-                if ((reqGroup == null || reqGroup.equals(tag.getChildValue("groupId").orElse(null))) &&
-                    resolvedPlugin.getArtifactId().equals(tag.getChildValue("artifactId").orElse(null)) &&
-                    (reqVersion == null || reqVersion.equals(tag.getChildValue("version").orElse(null)))) {
-                    return resolvedPlugin;
-                }
+        return findPlugin(tag, getResolutionResult().getPom().getPlugins());
+    }
+
+    public @Nullable Plugin findManagedPlugin(Xml.Tag tag) {
+        return findPlugin(tag, getResolutionResult().getPom().getPluginManagement());
+    }
+
+    private static @Nullable Plugin findPlugin(Xml.Tag tag, List<Plugin> pluginManagement) {
```

Comment:
Should maybe rename the `pluginManagement` parameter here.

Reply:
Thanks! Changed just now.